File: erykandroidcommon/src/main/java/pl/edu/amu/wmi/erykandroidcommon/verify/VerificationResult.java
package pl.edu.amu.wmi.erykandroidcommon.verify;

public class VerificationResult {

    private final String errorMessage;
    private final boolean failure;

    public VerificationResult() {
        this.failure = true;
        this.errorMessage = null;
    }

    public VerificationResult(String errorMessage) {
        this.errorMessage = errorMessage;
        this.failure = true;
    }

    public String getErrorMessage() {

        return errorMessage;
    }

    public boolean isFailure() {
        return failure;
    }

}

File: src/main/java/pl/edu/amu/wmi/erykandroidcommon/verify/VerificationResult.kt
package pl.edu.amu.wmi.erykandroidcommon.verify

class VerificationResult {

    val errorMessage: String?

    val isFailure: Boolean

    constructor() {
        this.isFailure = true
        this.errorMessage = null
    }

    constructor(errorMessage: String) {
        this.errorMessage = errorMessage
        this.isFailure = true
    }
}
